chain/chain/src/runtime/mod.rs

```diff
@@ -75,6 +76,7 @@ pub struct NightshadeRuntime {
     epoch_manager: Arc<EpochManagerHandle>,
     gc_num_epochs_to_keep: u64,
     state_parts_compression_lvl: i32,
+    is_cloud_archival_writer: bool,
```

Comment:
Alternatively, I could check for the existence of `CLOUD_HEAD_KEY` in `get_gc_stop_height_impl` to determine whether it `is_cloud_archival_writer`. However, this would block GC if `cloud_archival_writer` were removed from the config, since the `CLOUD_HEAD_KEY` would still remain in the database. We could implement automatic removal of `CLOUD_HEAD_KEY` when `cloud_archival_writer` is no longer present in the config. Still, I prefer the current approach as it is much simpler.